hw3/03-react-app/src/houses.js

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useRef } from 'react';
+import axios from 'axios';
+import { Chart, DoughnutController, ArcElement, Legend } from 'chart.js';
+import './houses.css';
+
+const backgroundColors = [
+  'rgba(54, 162, 235, 0.8)',
+  'rgba(255, 206, 86, 0.8)',
+  'rgba(255, 99, 132, 0.8)',
+  'rgba(75, 192, 192, 0.8)',
+  'rgba(153, 102, 255, 0.8)',
+  'rgba(255, 159, 64, 0.8)',
+  'rgba(199, 199, 199, 0.8)',
+  'rgba(83, 102, 255, 0.8)',
+  'rgba(40, 159, 64, 0.8)',
+  'rgba(210, 199, 199, 0.8)',
+  'rgba(78, 52, 199, 0.8)',
+];
+
+const borderColors = [
+  'rgba(54, 162, 235, 1)',
+  'rgba(255, 206, 86, 1)',
+  'rgba(255, 99, 132, 1)',
+  'rgba(75, 192, 192, 1)',
+  'rgba(153, 102, 255, 1)',
+  'rgba(255, 159, 64, 1)',
+  'rgba(159, 159, 159, 1)',
+  'rgba(83, 102, 255, 1)',
+  'rgba(40, 159, 64, 1)',
+  'rgba(210, 199, 199, 1)',
+  'rgba(78, 52, 199, 1)',
+];
+
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+const getColors = (dataArray, colorArray) => {
+  const repeatedColors = [];
+  dataArray.forEach((item, index) => {
+    repeatedColors.push(colorArray[index % colorArray.length]);
+  });
+  return repeatedColors;
+};
+
+export default function Houses() {
+  const donutChartRef = useRef(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(url);
```

Comment:
a small suggestion would be to fetch the API once in the App.js file and pass it to the other files as props.

Reply:
Thanks for the feedback. I made the changes in the updated PR